In [22]:
import os 
import csv 
import pandas as pd
import numpy as np
from pathlib import Path


In [23]:
data = ('2019_Winter_Data_Science_Intern_Challenge_Data_Set.csv')
csv_path = data

# Import the data set as a DataFrame
data_df = pd.read_csv(csv_path, encoding="utf-8")

In [24]:
# Incorrect calculation explained as a mistaken count() instead of sum() function
orderAmount_count = data_df['order_amount'].count()
totalItem_count = data_df['total_items'].count()
AOV = orderAmount_count / totalItem_count
print(AOV)
print("Incorrect as it just counts the total numbers of rows in each column")
orderAmount_sum = data_df['order_amount'].sum()
totalItem_count = data_df['total_items'].count()
AOV = orderAmount_sum / totalItem_count
print("%.2f" % AOV)
print("Incorrect count() is still used instead of sum() function")
print("This is the mistake done as described in the assignment")

1.0
Incorrect as it just counts the total numbers of rows in each column
3145.13
Incorrect count() is still used instead of sum() function
This is the mistake done as described in the assignment


Q1 a) Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 

Solution: The incorrect value of 3145.13 was resulted by taking the count function of total_items column which resulted in producing the total numbers of rows presented in the coloumn instead of telling the true value which we wanted that is the total number of items ordered. This caused the value of total_item to significantly low (error). So when the resulting AOV is calulated by using the formula total order amount / total number of items order. The value of AOv increased greatly resulting in the wrong value. 

Q1 b) What metric would you report for this dataset?

Solution: I want to report the sums of both 'order_amount' and 'total_items' to calculate the Average of the values by totalling the numbers in each column instaed of calculating the total rows 

In [25]:
#Correct calculation
orderAmount_sum = data_df['order_amount'].sum()
print(orderAmount_sum)
totalItem_sum = data_df['total_items'].sum()
print(totalItem_sum)

15725640
43936


In [26]:
AOV = orderAmount_sum/totalItem_sum
print("%.2f" % AOV)

357.92


Q1 c) What is its value?

Solution: 357.92